lrpar/src/lib/parser.rs

```diff
@@ -289,9 +300,15 @@ where
                     let prior = *pstack.last().unwrap();
                     pstack.push(self.stable.goto(prior, ridx).unwrap());
 
+                    let tail = span[span.len() - 1];
```

Comment:
What is the purpose if this? I can't really tell just from the code.

Reply:
This code is actually where most of the drama when creating this PR happened! Here's an easier rewrite:

```rust
                    let sp = (span[pop_idx - 1], span[span.len() - 1]);
                    if pop_idx < span.len() - 1 {
                        span.drain(pop_idx..span.len() - 1);
                    }
```

Basically we want to delete from `pop_idx..span.len() - 1` *but* sometimes `popidx == span.len() - 1` and then `drain` complains about that (why... I don't know). Unfortunately there isn't a `remove_range` function, hence the use of `drain`.

Reply:
Ah! That makes sense.